src/transformers/models/maskformer/feature_extraction_maskformer.py

```diff
@@ -0,0 +1,566 @@
+        """
+        if not (masks.shape[0] == scores.shape[0] == labels.shape[0]):
+            raise ValueError("mask, scores and labels must have the same shape!")
+
+        to_keep = labels.ne(num_labels) & (scores > object_mask_threshold)
+
+        return masks[to_keep], scores[to_keep], labels[to_keep]
+
+    def post_process_semantic_segmentation(
+        self, outputs: "MaskFormerForInstanceSegmentationOutput", target_size: Tuple[int, int] = None
+    ) -> "torch.Tensor":
+        """
+        Converts the output of [`MaskFormerForInstanceSegmentationOutput`] into semantic segmentation predictions. Only
+        supports PyTorch.
+
+        Args:
+            outputs ([`MaskFormerForInstanceSegmentationOutput`]):
+                The outputs from [`MaskFormerForInstanceSegmentation`]
+
+        Returns:
+            `torch.Tensor`: A tensor of shape `batch_size, height, width`
+        """
+        segmentation = self.post_process_segmentation(outputs, target_size)
+        semantic_segmentation = segmentation.argmax(dim=1)
+        return semantic_segmentation
+
+    def post_process_panoptic_segmentation(
+        self,
+        outputs: "MaskFormerForInstanceSegmentationOutput",
+        object_mask_threshold: float = 0.8,
+        overlap_mask_area_threshold: float = 0.8,
+        is_thing_map: Optional[Dict[int, bool]] = None,
+    ) -> List[Dict]:
+        """
+        Converts the output of [`MaskFormerForInstanceSegmentationOutput`] into image panoptic segmentation
+        predictions. Only supports PyTorch.
+
+        Args:
+            outputs ([`MaskFormerForInstanceSegmentationOutput`]):
+                The outputs from [`MaskFormerForInstanceSegmentation`]
+            object_mask_threshold (`float`, *optional*, defaults to `0.8`):
+                The object mask threshold.
+            overlap_mask_area_threshold (`float`, *optional*, defaults to `0.8`):
+                The overlap mask area threshold.
+            is_thing_map (`Dict[int, bool]`, *optional*, defaults to `None`):
+                Dictionary mapping class indices to either True or False, depending on whether or not they are a thing.
+                If not set, defaults to the `is_thing_map` of ADE20K-150 panoptic.
+
+        Returns:
+            `List[Dict]`: A list of dictionaries, one per image, each dictionary containing two keys:
+            - **segmentation** -- a tensor of shape `(height, width)` where each pixel represent a `segment_id`.
+            - **segments** -- a dictionary with the following keys
+                - **id** -- an integer representing the `segment_id`.
+                - **category_id** -- an integer representing the segment's label.
+                - **is_thing** -- a boolean, `True` if `category_id` was in `is_thing_map`, `False` otherwise.
+        """
+
+        if is_thing_map is None:
+            # default to is_thing_map of COCO panoptic
+            is_thing_map = {i: i <= 90 for i in range(201)}
```

Comment:
This seems rather dangerous as a user that didn't set anything will end up with something completely unrelated to their problem. Should this be a required argument in the init?

In any case, it seems like it should be saved along the feature extractor for the inference widgets/pipeline use of the model.

Reply:
I agree, this is also present in [detr](https://github.com/huggingface/transformers/blob/38bed912e36e1725ede1f0e8c61a514f378697c3/src/transformers/models/detr/feature_extraction_detr.py#L831)

Reply:
That does not mean we should replicate here. We can port the fix on Detr once this PR is merged :-)

Reply:
I am not 100% how to precede, on one hand, we should save it but on the other hand, this is an argument only used in one function. What if I log a warning message? "e.g. `is_thing_map not settled, initialized to COCO`

Reply:
Yes you can log a warning.